client/python/openlineage/client/generated/base.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2018-2024 contributors to the OpenLineage project
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import ClassVar
+
+from attr import define, field
+from openlineage.client.constants import DEFAULT_PRODUCER
+from openlineage.client.utils import RedactMixin
+
+
+@define(kw_only=True)
+class BaseEvent(RedactMixin):
+    eventTime: str = field()  # noqa: N815
+    """the time the event occurred at"""
+
+    producer: str = field(default="", kw_only=True)
+    schemaURL: str = field(  # noqa: N815
+        default="https://openlineage.io/spec/2-0-2/OpenLineage.json#/$defs/BaseEvent", init=False
+    )
+    _base_skip_redact: ClassVar[list[str]] = ["producer", "schemaURL"]
+    _additional_skip_redact: ClassVar[list[str]] = []
+
+    def __attrs_post_init__(self) -> None:
+        import warnings
+
+        if self.producer:
+            warnings.warn(
+                "producer as argument is deprecated. Please use `set_producer` instead.",
+                DeprecationWarning,
+                stacklevel=2,
+            )
+        else:
+            self.producer = PRODUCER
```

Comment:
```suggestion
        if not self.producer:
            self.producer = PRODUCER
```
i think we should allow setting the producer with argument.

Reply:
We can always deprecate this later if we decide so. I'll remove deprecation warning and adjust logic.

Reply:
I changed the logic and added test for it. @kacpermuda  would you mind taking a look into this one more time?

Reply:
lgtm